File: 169-MajorityElement/169-MajorityElement.java
// Last updated: 8/1/2025, 10:39:08 PM
class Solution {
    public int majorityElement(int[] nums) {
      return mooreVoting(nums);
        // mooore voting
        
    }
   public static int mooreVoting(int[] arr) {
		int e=arr[0];
        int n=arr.length;
		int v=1;
		for(int i=1;i<n;i++) {
			if(arr[i]==e) {
				v++;
			}
			else {
				v--;
				if(v==0) {
					e=arr[i];
					v=1;
				}
			}
		}
		return e;

   }
}

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 8/1/2025, 10:39:15 PM
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
       n=len(nums)
       d={}
       for i in nums:
         if i in d:
          d[i]+=1
         else:
           d[i]=1
       for i, ns in d.items():
               if ns > n/2:
                 return i
  